### PR TITLE
fix(Message): Message global static method 'config' setting duration is invalid

### LINKED ISSUE #40223: message.config中的自动关闭延时 duration 设置无效
### Reproduction link

[https://stackblitz.com/edit/react-ts-xbbbgf?file=App.tsx](https://stackblitz.com/edit/react-ts-xbbbgf?file=App.tsx)

### Steps to reproduce

打开链接，点击弹出message，时间设置了100秒，3秒就关闭了

### What is expected?

duration 能够全局设置

### What is actually happening?

duration 全局设置无效

| Environment | Info |
| --- | --- |
| antd | 5.1.4 |
| React | react 18 |
| System | Windows 10 |
| Browser | Chrome 104.0.5112.102 |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/40223

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix `message` global static method `config` setting duration is invalid          |
| 🇨🇳 Chinese | 修复 `message` 组件通过`config`设置`duration`无效问题         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
1. ci failed, please check it
2. could you add a test case for it?

**Comment by Yuiai01** [comment]
> 1. ci failed, please check it
> 2. could you add a test case for it?

好的大佬，关于这两个问题：
1. ci failed 已经修复；
2. 之前已经存在针对 config 中 duration 的测试用例，所以我基于之前的用例进行了更改。

File: components/message/__tests__/config.test.tsx [hunk 1]
```diff
@@ -104,10 +104,15 @@ describe('message.config', () => {
     expect(document.querySelectorAll('.ant-message-notice')).toHaveLength(1);
 
     act(() => {
-      jest.advanceTimersByTime(100);
+      jest.advanceTimersByTime(4000);
     });
+
     expect(document.querySelectorAll('.ant-message-notice')).toHaveLength(1);
 
+    await triggerMotionEnd('.ant-message-notice');
```
Comment (zombieJ) [inline components/message/__tests__/config.test.tsx:112]: 有点 tricky，按理说 5s 没到应该还是没有关闭。但是这里因为 triggerMotionEnd 触发了移除。在这行上面再加个 `jest.advanceTimersByTime(2000);` 过一下免得以后卡不懂。

Comment (Yuiai01) [reply components/message/__tests__/config.test.tsx:112]: > 有点 tricky，按理说 5s 没到应该还是没有关闭。但是这里因为 triggerMotionEnd 触发了移除。在这行上面再加个 `jest.advanceTimersByTime(2000);` 过一下免得以后卡不懂。

嗯嗯，大佬我之前写的是 `jest.advanceTimersByTime(5500);` 然后判断获取到的元素数组长度的，但是不管怎么加`advanceTimersByTime` 中的时间，这个长度依旧是 1，就很奇怪，能麻烦大佬指教一下吗。😭

Comment (Yuiai01) [reply components/message/__tests__/config.test.tsx:112]: image
image

**Review by zombieJ — APPROVED** [review]

**Pull request merged**